backend/lucene/src/main/java/org/hibernate/search/backend/lucene/types/projection/impl/LuceneGeoPointFieldProjectionBuilderFactory.java

```diff
@@ -44,15 +42,15 @@ public LuceneGeoPointFieldProjectionBuilderFactory(boolean projectable,
 	@Override
 	@SuppressWarnings("unchecked") // We check the cast is legal by asking the converter
 	public <T> FieldProjectionBuilder<T> createFieldValueProjectionBuilder(String absoluteFieldPath,
-			Class<T> expectedType) {
+			Class<T> expectedType, ProjectionConverter projectionConverter) {
 		checkProjectable( absoluteFieldPath, projectable );
 
-		if ( !converter.isConvertedTypeAssignableTo( expectedType ) ) {
+		if ( projectionConverter.isEnabled() && !converter.isConvertedTypeAssignableTo( expectedType ) ) {
```

Comment:
You should check whether the type is assignable in all cases. Just, when the projection converter is disabled, you will call `isConvertedTypeAssignableTo` on the *raw* converter.

Note that your tests didn't fail, so you're missing a test where you try to pass the wrong type to `rawField(String, Class)`.